aiida_vasp/workchains/vasp.py

```diff
@@ -375,29 +418,105 @@ def _handle_unfinished_calculation(self, node):
         if incar.get('nsw', -1) > 0:
             if 'structure' not in node.outputs.structure:
                 self.report('Performing a geometry optimisation but the output structure is not found.')
-                return ProcessHandlerReport(do_break=True, exit_code=self.exit_codes.ERROR_MANUAL_INTERVENTION_NEEDED)  #pylint: disable=no-member
+                return ProcessHandlerReport(
+                    do_break=True,
+                    exit_code=self.exit_codes.ERROR_MANUAL_INTERVENTION_NEEDED.format(message='No output structure for restart.'))  #pylint: disable=no-member
             self.report('Continuing geometry optimisation using the last geometry.')
             self.ctx.inputs.structure = node.outputs.structure
-            self._attach_wavecar(node)
+            self._setup_restart(node)
         else:
             # Single point calculation - attach the wavecar if possible
-            has_wavecar = self._attach_wavecar(node)
+            has_wavecar = self._setup_restart(node)
             if not has_wavecar:
                 # If there is no WAVECAR there is not much we can do.....
                 self.report('The calcualtions did not finish and no WAVECAR is avalaible for restart.')
-                return ProcessHandlerReport(do_break=True, exit_code=self.exit_codes.ERROR_MANUAL_INTERVENTION_NEEDED)  #pylint: disable=no-member
-        return ProcessHandlerReport()
-
-    def _attach_wavecar(self, node):
+                return ProcessHandlerReport(
+                    do_break=True,
+                    exit_code=self.exit_codes.ERROR_MANUAL_INTERVENTION_NEEDED.format(message='No WAVECAR for continuation.'))  #pylint: disable=no-member
+        return ProcessHandlerReport(do_break=True)
+
+    @process_handler(priority=80, exit_codes=[VaspCalculation.exit_codes.ERROR_ELECTRONIC_NOT_CONVERGED])
+    def _handle_electronic_converge_problem(self, node):
```

Comment:
> Great addition. Can we update this to follow https://www.vasp.at/wiki/images/b/b6/VASP_lecture_Basics2.pdf page 28? Also add that we drop down to ALGO=N as a first step. I can fix it if you are okey with me pushing to your PR?

I actually don't think we should follow that guide, apart from the drop to N part (only if from Fast!)... That approach would for very difficult systems, but that is beyond the scope of this simple handle. And 'play with mixing parameters' is rather vague... The goal here to fix simple problems or those that just happen by chance to improve the robustness, such as the NELM is too low, using too aggressive mixing parameters/convergence criteria e.t.c. Otherwise, the user needs to intervene.

Reply:
>Having said that, maybe there is a tool to take a code snippet and create a workflow diagram? That would be useful and make sure our variable names are 1:1.

Yeah I will look into that, the priority and emitting/not emitting the report at each handler is rather abstract....

Reply:
> I actually don't think we should follow that guide, apart from the drop to N part (only if from Fast!)... That approach would for very difficult systems, but that is beyond the scope of this simple handle. And 'play with mixing parameters' is rather vague... The goal here to fix simple problems or those that just happen by chance to improve the robustness, such as the NELM is too low, using too aggressive mixing parameters/convergence criteria e.t.c. Otherwise, the user needs to intervene.

Yes, so now we are entering something I discussed in detail with Sebastiaan. Where do we draw the line between handlers and more workchain type of fixing and making calculations tractable? I think we need to agree on this. In fact, I would possibly like to avoid putting these things into handlers at all. Such things can be made in its own workchain, with its own provenance etc. So I would honestly suggest that we do not handle these things at the handler level. Also, if we have this "simple" approach, and then we add a workchain on top that does a more "complete" approach it would be confusing to the user. Maybe we can leave it here and I will lift these things out into a workchain later?

Reply:
OK. I think it is important to have this functionality to improve the robustness and lifts the burden of higher-level workchains, otherwise, there is hard to justify having all these frameworks but not have simple fixes implemented. The provenance may not be clear on the graph, but everything is there in the database. 

I would like to leave them here, and if later a higher level workchain is developed that can address the provenance in a better way, the handlers here can always be disabled at the run time.

Reply:
Great. Agreed.